### ISSUE TITLE
[BUG] conflicting transactions fail to execute when submitted in parallel.

### ISSUE THREAD

**Issue by sdjasj** [sei-protocol/sei-chain#1430]
**Seid version**
```
name: sei
server_name: <appd>
version: v4.1.4-evm-devnet
commit: 6e26498471079268799dd14091ca7256680cb82f
build_tags: netgo ledger,
go: go version go1.22.0 linux/amd64
build_deps:
- cosmossdk.io/errors@v1.0.0
- filippo.io/edwards25519@v1.0.0-rc.1
- github.com/99designs/keyring@v1.2.1
```

**Chain ID**
Local 4-docker-node chain with EVM parallel mechanism

**Describe the bug**
Transactions can be executed normally when committed serially, but most transactions fail when committed in parallel.

**To Reproduce**
1.run a 4-docker-node private chain locally.
2.Deploy some Solidity contracts through Forge, where the functions of the contracts conflict in accessing state variables.
3.Use cast to send transactions one by one; transactions can be successfully sent and executed.
4.Use cast to send transactions in parallel; most transactions fail to execute.

**Expected behavior**
When transactions are submitted in parallel, all transactions can be successfully executed.

**Screenshots**
When transactions are submitted in parallel, some transactions fail to execute.
![957f30f9b1261ec4c8e67764a421803](https://github.com/sei-protocol/sei-chain/assets/89853352/ed7b3435-cba5-4062-8bdf-bf8722a4ce1f)

**Additional context**

**Comment by philipsu522** [comment]
hey @sdjasj we've fixed the evm logic quite a bit. If you're still having issues let's track in a new issue, thanks!
